types.cc

```diff
@@ -1401,16 +1401,15 @@ tuple_type_impl::get_instance(std::vector<data_type> types) {
     return intern::get_instance(std::move(types));
 }
 
-static void validate_aux(const tuple_type_impl& t, bytes_view v, cql_serialization_format sf) {
+template <FragmentedView View>
+static void validate_aux(const tuple_type_impl& t, View v, cql_serialization_format sf) {
     auto ti = t.all_types().begin();
-    auto vi = tuple_deserializing_iterator::start(v);
-    auto end = tuple_deserializing_iterator::finish(v);
```

Comment:
Can `tuple_deserializing_iterator` be removed now?

Reply:
Not yet. However, there are only two other uses in the codebase, and they can be trivially replaced with `read_tuple_element()`, so we can remove it soon. But since those uses are unrelated, let's do it in another patch.

Reply:
ok